packages/beacon-transport-walletconnect/src/communication-client/WalletConnectCommunicationClient.ts

```diff
@@ -428,40 +435,24 @@ export class WalletConnectCommunicationClient extends CommunicationClient {
     this.notifyListeners(session, acknowledgeResponse)
   }
 
-
   private async updateActiveAccount(namespaces: SessionTypes.Namespaces) {
     try {
       const accounts = this.getTezosNamespace(namespaces).accounts
       if (accounts.length === 1) {
-        // TODO: Use public key from namespace. Use getAddressFromPublicKey(pk)
-        const [_namespace, chainId, address] = accounts[0].split(':', 3)
-        this.activeAccount = address
-        this.activeNetwork = chainId
-
-        const signClient = await this.getSignClient()
+        const [_namespace, chainId, addressOrPbk] = accounts[0].split(':', 3)
         const session = this.getSession()
-
         let publicKey: string | undefined
 
-        // Get the public key from the wallet
-        const result = await signClient.request<
-          [
-            {
-              algo: 'ed25519'
-              address: string
-              pubkey: string
-            }
-          ]
-        >({
-          topic: session.topic,
-          chainId: `${TEZOS_PLACEHOLDER}:${chainId}`,
-          request: {
-            method: PermissionScopeMethods.GET_ACCOUNTS,
-            params: {}
-          }
-        })
+        if (addressOrPbk.startsWith('edpk')) {
+          publicKey = addressOrPbk
+          this.activeAccount = await getAddressFromPublicKey(publicKey)
+        } else {
+          this.activeAccount = addressOrPbk
+          this.activeNetwork = chainId
```

Comment:
`this.activeNetwork` should be set regardless of whether `addressOrPbk` is a public key or address

Reply:
Hi @jsamol, I moved up `this.activeAccount = addressOrPbk` in the outer scope